internal/docstore/driver/driver.go

```diff
@@ -18,23 +18,35 @@ package driver // import "gocloud.dev/internal/docstore/driver"
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"gocloud.dev/internal/gcerr"
 )
 
 // A Collection is a set of documents.
 type Collection interface {
-	// RunActions executes a sequence of actions.
-	// Implementations are free to execute the actions however they wish, but it must
-	// appear as if they were executed in order. The actions need not happen
-	// atomically.
+	// RunActions executes a slice of actions.
 	//
-	// RunActions should return immediately after the first action that fails.
-	// The first return value is the number of actions successfully
-	// executed, and the second is the error that caused the action to fail.
+	// If unordered is false, it must appear as if the actions were executed in the
+	// order they appear in the slice, from the client's point of view. The actions
+	// need not happen atomically, nor does eventual consistency in the provider
+	// system need to be taken into account. For example, after a write returns
+	// successfully, the driver can immediately perform a read on the same document,
+	// even though the provider's semantics does not guarantee that the read will see
+	// the write. RunActions should return immediately after the first action that fails.
```

Comment:
But, IIUC, it needs to fill in `ErrorNotExecuted` for all the subsequent actions? Maybe this is something we could do in the portable type (for ordered mode only; for unordered mode I think the driver needs to tell us all the info).

Oh I now see you mentioned this below. Maybe it would be more clear to make less optional; e.g., "in ordered mode, just return the first error and its position" (and the portable type will in the `ErrNotExecuted` for the rest); "in unordered mode, always return n errors"?

Reply:
Sorry you had to puzzle through that. It was obsoleted by `ActionListError`. Originally RunActions just returned a `[]error`, which required all these special cases and conventions.